agent/gaeul-agent.c

```diff
@@ -411,7 +427,8 @@ _edge_user_command_cb (ChamgeEdge * edge, const gchar * user_command,
           if (json_object_has_member (json_object, "params")) {
```

Comment:
I think we should change the logic here a bit. We now have

```
if (!self->is_playing) {
  if (_get_srt_uri (self) == CHAMGE_RETURN_OK) {
    ... parse parameters from streamingStart message ...

    g_idle_add ((GSourceFunc) _start_pipeline, self);
    g_debug ("streaming is starting");
  }
}
```

But since we now can get the target URI for streaming from the message, not being able to obtain it using `_get_srt_uri()` (which reads it from GSettings) shouldn't prevent us from parsing the message (and reading the actual URI from `"params"`).

I suggest:

```
if (!self->is_playing) {
  ... parse parameters from streamingStart message ...

  if (!self->srt_target_uri) {
    /* No URI in the message - try reading the default value from gsettings */
    _get_srt_uri (self)
  }

  if (self->srt_target_uri) {
    g_idle_add ((GSourceFunc) _start_pipeline, self);
    g_debug ("streaming is starting");
  } else {
    g_warning ("Couldn't determine SRT URI for streaming");
  }
}
```

Reply:
I tried to keep the logic and override the `self->srt_target_uri` only if `uri` parameter is found in the message.
Please confirm if that is OK or it needs to be changed, maybe I'm missing something.

Reply:
I think the current logic doesn't make much sense anymore. If you don't have any default URI in GSettings, you never get to parsing the URI from the message and get completely stuck.

In my opinion the message parameter has higher priority. We should use the default only when we're not told otherwise in `streamingStart` msg.

Reply:
I see your point, thanks for sharing this. I'll add the fix.